modules/build/src/main/scala/scala/build/preprocessing/directives/UsingResourcesDirectiveHandler.scala

```diff
@@ -43,12 +47,16 @@ case object UsingResourcesDirectiveHandler extends UsingDirectiveHandler {
     cwd: ScopePath,
     positionOpt: Option[Position]
   ): Either[BuildException, BuildOptions] = either {
-    val root   = value(Directive.osRoot(cwd, positionOpt))
-    val paths  = DirectiveUtil.stringValues(values)
-    val paths0 = paths.map(os.Path(_, root))
+    val (virtualRootOpt, rootOpt) = Directive.osRootResource(cwd)
+    val paths                     = DirectiveUtil.stringValues(values)
+    val paths0                    = rootOpt.map(root => paths.map(os.Path(_, root)))
+    val virtualPaths = virtualRootOpt.map(virtualRoot =>
+      paths.map(path => virtualRoot / os.SubPath(path)).map(_.toString())
+    )
     BuildOptions(
       classPathOptions = ClassPathOptions(
-        extraClassPath = paths0
+        extraClassPath = paths0.toList.flatten,
+        resourceVirtualDir = virtualPaths.toList.flatten
```

Comment:
NIT Why not keeping `os.SubPath`s here? It seems they are parsed again as `os.SubPath` above.

Reply:
I'm not very familiar with shapeless, but  if I use `os.SubPath` in `ClassPathOptions.resourceVirtualDir : Seq[os.SubPath]`, then it throw compilation error:

```
[error] /Users/lwronski/projects/scala-cli/modules/build/src/main/scala/scala/build/options/ClassPathOptions.scala:18:73: could not find implicit value for parameter gen: shapeless.LabelledGeneric.Aux[T,R]
[error]   implicit val hasHashData: HasHashData[ClassPathOptions] = HasHashData.derive
```


Unfortunately, I didn't fix it, so I used `Seq [String]` instead of `Seq[os.SubPaths]`

Reply:
It's a matter of having an implicit `HashedType` instance for `os.SubPath` I think. Something like this in the `HashedType` companion should do:
```scala
implicit val subPath: HashedType[os.SubPath] = {
  path => path.toString
}
```